src/components/Identity/syncDeclaredIds.js

```diff
@@ -0,0 +1,52 @@
+import { sha256Buffer, normalizeCustomerIds, bufferToHex } from "./util";
+import { crc32 } from "../../utils";
+import { COOKIE_NAMES } from "./constants";
+
+const { CUSTOMER_ID_HASH } = COOKIE_NAMES;
+
+const makeServerCall = (payload, lifecycle, network) => {
+  return lifecycle.onBeforeDataCollection(payload).then(() => {
+    return network.sendRequest(payload, payload.expectsResponse);
+  });
+};
+
+export default (
+  customerIds,
+  event,
+  payload,
+  cookie,
+  lifecycle,
+  network,
+  optIn
+) => {
+  const normalizedCustomerIds = normalizeCustomerIds(customerIds);
+  const idNames = Object.keys(normalizedCustomerIds);
+  const idsToHash = idNames.filter(idName => customerIds[idName].hash);
+  const hashedIds = idsToHash.map(id =>
+    sha256Buffer(normalizedCustomerIds[id].id)
+  );
+
+  Promise.all(hashedIds).then(values => {
```

Comment:
(continued from https://github.com/adobe/alloy/pull/167/files#r315935018) ...then rename `values` to `hashedIds` to be more specific.